src/client.ts

```diff
@@ -326,13 +338,11 @@ class OptimizelyReactSDKClient implements ReactSDKClient {
       return;
     }
 
-    if (!this.userContext || (this.userContext && !areUsersEqual(userInfo, this.user))) {
```

Comment:
Why do we remove this if-condition? It'll be useful for avoiding the unnecessary state change.

Reply:
In the calling method, we're updating the React-side `this.user` to match the user-provided `userInfo` input. 

This method is responsible for updating the underlying JS SDK userContext. Here we've been comparing the provided `userInfo` and the (now updated) React `this.user` which skips updating the underlying JS userContext.

The appropriate SRP would be to read the underlying JS `userContext` and then compare it with the user-provided `userInfo` to see if it needs updating.  

I can do that.